wgpu-hal/src/metal/adapter.rs

```diff
@@ -952,6 +960,8 @@ impl super::PrivateCapabilities {
                 max_vertex_buffers: self.max_vertex_buffers,
                 max_vertex_attributes: 31,
                 max_vertex_buffer_array_stride: base.max_vertex_buffer_array_stride,
+                min_subgroup_size: 4,
```

Comment:
Sigh. All Apple Silicon devices have a fixed subgroup size of 32, though it's plausible that this may change in the future. I really wish there were a principled way to set this; it's too tempting to just sniff the device name.

Reply:
You are forgetting about quad operations. According to the MSL spec:
> A quad-group is a SIMD-group with the thread execution width of 4.

Anyway, any suggested change here code wise?